src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -195,69 +201,269 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
-            }
-            else
+            using var directory = new TempDirectory();
+            string path = Path.Combine(directory.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
+            DirectoryInfo info = new DirectoryInfo(path);
+            DirectorySecurity security = new DirectorySecurity();
+
+            Assert.Throws<UnauthorizedAccessException>(() =>
             {
-                Assert.Throws<ArgumentNullException>(() => info.Create(null));
-            }
+                if (PlatformDetection.IsFullFramework)
+                {
+                    FileSystemAclExtensions.Create(info, security);
+                }
+                else
+                {
+                    info.Create(security);
+                }
+            });
         }
 
         [Fact]
-        public void DirectoryInfo_Create_NotFound()
+        public void DirectoryInfo_Create_DefaultDirectorySecurity()
         {
-            DirectoryInfo info = new DirectoryInfo(@"W:\\I\\Do\\Not\\Exist");
             DirectorySecurity security = new DirectorySecurity();
-            Assert.Throws<DirectoryNotFoundException>(() => info.Create(security));
+            VerifyDirectorySecurity(security);
         }
 
         [Theory]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadAndExecute, AccessControlType.Deny)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.WriteData, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.WriteData, AccessControlType.Deny)]
         [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadData, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Deny)]
         [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Deny)]
         public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(
             WellKnownSidType sid,
             FileSystemRights rights,
             AccessControlType controlType)
         {
-
             DirectorySecurity security = GetDirectorySecurity(sid, rights, controlType);
             VerifyDirectorySecurity(security);
         }
 
         #endregion
 
+        #region FileInfo Create
+
+        [Fact]
+        public void FileInfo_Create_NullFileInfo()
+        {
+            FileInfo info = null;
+            FileSecurity security = new FileSecurity();
+
+            Assert.Throws<ArgumentNullException>("fileInfo", () =>
+            {
+                if (PlatformDetection.IsFullFramework)
```

Comment:
Nit, there's a bunch of repetition among these tests, you could do

```c#

private void FileInfoCreate(info, mode, rights, share, buffer, options, security)
{
                if (PlatformDetection.IsFullFramework)
                {
                    FileSystemAclExtensions.Create(....);
                }
                else
                {
                    info.Create(...);
                }
}
```
Then your test would be just eg
```c#
        [Fact]
        public void FileInfo_Create_NullFileSecurity()
        {
            FileInfo info = new FileInfo("path");

            Assert.Throws<ArgumentNullException>("fileSecurity", () => FileInfoCreate(info, FileMode.Create, FileSystemRights.WriteData, FileShare.Read, DefaultBufferSize, FileOptions.None, null);
        }
```

Reply:
Or - another idea. You could have a separate source file that's only included when building tests for .NET Framework which defines your new extension method and just passes it through to the .NET Framework method. That might be nice because the tests themselves don't need to know it exists. They just call the .NET Core method.

If it's worth doing, you coudl do it later.

Reply:
Generally you want tests to have as little repetition as possible, because that makes it easier to see what changes between them (what the test cases are)

Reply:
Totally understood, @danmosemsft . Would it be ok if I address this in a separate PR so I get this merged today?

Reply:
good call